crypto/bn/bn_local.h

```diff
@@ -35,6 +35,8 @@
 /* #define BN_DEBUG */
 /* #define BN_DEBUG_RAND */
 
+#define BN_SIZE_LIMIT           (4096 / BN_BYTES)
```

Comment:
BN_CONSTTIME_SIZE_LIMIT or BN_MONT_CONSTTIME_SIZE_LIMIT?

Reply:
BN_SIZE_LIMIT feels right, as @davidben pointed out also BN_mod_exp and BN_mod_exp_mont will eventually call
bn_mul_mont_fixed_top and thus bn_mul_mont, so it is neither specific to consttime nor montgomery at all.
Actutally the whole issue starts to look a lot less harmless than in the beginning.

Reply:
Then BN_MONT_SIZE_LIMIT?

Reply:
Unless this limit applies to all BN operations it should not be called BN_SIZE_LIMIT.

Reply:
It is more the upper bound on the stack allocation of any BN operation we are willing to allow.
It also affects BN_mod_exp and BN_mod_sqrt.
Note: there are many call sites of the affected functions, especially of BN_mod_exp,
but not all of them seem to sanity check the input parameters.
If the modulus is a large enough odd number, a stack overflow will probably happen unexpectedly